code/game/machinery/vending.dm

```diff
@@ -732,7 +733,7 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 					/obj/item/weapon/lighter/greyscale = 4,
 					/obj/item/weapon/storage/fancy/rollingpapers = 5)
 	contraband = list(/obj/item/weapon/lighter = 3)
-	premium = list(/obj/item/weapon/storage/fancy/cigarettes/cigpack_robustgold = 3, \
+	premium = list(/obj/item/weapon/storage/fancy/cigarettes/cigpack_robustgold = 1, \

```

Comment:
since you're updating the amount of products in the vending machine you need to update the amount in the resupply unit for that machine accordingly.